frontend/components/forms/checkbox/component.stories.tsx

```diff
@@ -1,43 +1,134 @@
-import { Story } from '@storybook/react/types-6-0';
+import React from 'react';
+
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+import { action } from '@storybook/addon-actions';
+import { Story, Meta } from '@storybook/react/types-6-0';
+
+import Button from 'components/button';
 
-import Checkbox from './component';
 import { CheckboxProps } from './types';
 
+import Checkbox from '.';
+
 export default {
-  title: 'Components/Forms/Checkbox',
   component: Checkbox,
+  title: 'Components/Forms/Checkbox',
   argTypes: {
-    theme: {
-      control: {
-        type: 'select',
-        options: ['dark', 'light'],
-      },
-    },
-    status: {
-      control: {
-        type: 'select',
-        options: ['none', 'valid', 'error', 'disabled'],
-      },
-    },
-    InputHTMLAttributes: {
-      name: 'InputHTMLAttributes',
-      description: 'https://www.w3schools.com/tags/tag_textarea.asp',
-      table: {
-        type: {
-          summary: 'InputHTMLAttributes',
-          detail: null,
-        },
-      },
-      control: {
-        disabled: true,
-      },
-    },
+    register: { control: { disable: true } },
   },
-};
+} as Meta;
+
+interface FormValues {
+  accept: string;
+}
+
+const Template: Story<CheckboxProps<FormValues>> = (args: CheckboxProps<FormValues>) => {
+  const { register } = useForm<FormValues>();
 
-const Template: Story<CheckboxProps> = (args) => <Checkbox {...args} />;
+  return (
+    <div className="p-4">
+      <Checkbox register={register} {...args} />
+      <p className="mt-2 text-xs text-gray-50">
+        This input is not accessible. Either define <code>aria-label</code> or associate a label
+        (see the “With Label” story).
+      </p>
+    </div>
+  );
+};
 
-export const Default = Template.bind({});
+export const Default: Story<CheckboxProps<FormValues>> = Template.bind({});
 Default.args = {
-  theme: 'dark',
+  id: 'form-accept',
+  name: 'accept',
+  registerOptions: {
+    disabled: false,
+  },
+};
+
+const TemplateWithLabel: Story<CheckboxProps<FormValues>> = (args: CheckboxProps<FormValues>) => {
```

Comment:
This story is not accessible. When you click on the label, the checkbox should get the focus and toggle. If you try in Storybook, you'll see it's not the case. Below I will detail why it's not working.